app/src/features/surveys/view/survey-spatial/components/animal/SurveySpatialAnimalCapturePopup.tsx

```diff
@@ -45,15 +49,17 @@ export const SurveySpatialAnimalCapturePopup = (props: ISurveySpatialAnimalCaptu
       closeButton={true}
       autoPan={true}
       eventHandlers={{
-        add: () => {
-          captureDataLoader.load(String(feature.id));
+        add: async () => {
+          const capture = await captureDataLoader.load();
+          // Fetch critter information to get the animal's nickname to display, which isn't included in the capture response
+          if (capture) animalDataLoader.load(capture.critter_id);
         }
       }}>
       <SurveyMapPopup
-        isLoading={captureDataLoader.isLoading || !captureDataLoader.isReady}
-        title="Capture"
-        metadata={captureDataLoader.data ? getCaptureMetadata(captureDataLoader.data) : []}
-        key={`capture-feature-popup-${feature.id}`}
+        isLoading={captureDataLoader.isLoading || animalDataLoader.isLoading}
+        title="Capture Details"
+        metadata={formatPopupMetadata()}
```

Comment:
(We probably have other places that do this too, as its not always something we look for)

It is technically more performant to run `formatPopupMetadata()` in advance, and just pass the result to this prop. Additionally, wrapping it in a `useMemo` will ensure that `formatPopupMetadata` only gets called if something has changed.

The way it is now, the function gets called on every render, even if the response is destined to be the same.

```
const popupMetadata = useMemo(() => {
  ...
}, [captureDataLoader.data, animalDataLoader.data]);

<SurveyMapPopup
  metadata={popupMetadata}
  ... 
/>
```

Reply:
Its quite possible that this component already doesn't re-render much and the performance is negligible, but its a good practice in general.